plugin/pkg/admission/forgiveness/admission.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package forgiveness
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	apierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+func init() {
+	admission.RegisterPlugin("DefaultForgivenessSeconds", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewDefaultForgivenessSeconds(client), nil
+	})
+}
+
+// plugin contains the client used by the admission controller
+// It will add default forgiveness tolerations for every pod
+// that tolerate taints `notready:NoExecute` and `unreachable:NoExecute`,
+// with forgiveness period of 5 minutes.
+// If the pod already specifies a toleration for taint `notready:NoExecute`
+// or `unreachable:NoExecute`, the plugin won't touch it.
+type plugin struct {
+	*admission.Handler
+	client clientset.Interface
+}
+
+var defaultForgivenessSeconds int64 = 300
```

Comment:
Can we make this an API server flag? It's basically serving the same purpose as nc.podEvictionTimeout today, so I think it should be flag-controlled.